File: Testing/basic_movement/moon_walk.cpp
//This file is a template for the moon_walk dance of the robot first dance function we created
#include "PinAssignments.h"
#include "Libraries.h"

Servo left1servo; // Offset +10
Servo left2servo;
Servo right1servo;  // Offset +5
Servo right2servo;

int startPos = 40;
int pos = startPos;    // variable to store the servo position
int endPos = 120;
int flatPos = 90;
int left1Offset = 10;
int left2Offset = 0;
int right1Offset = 5;
int right2Offset = 0;
int footTilt = 8;

/**
 * @brief setup motors for the moon_walk function
 * 
 */
void moon_walk_setup() {
  left1servo.attach(SERVO_LEFT_1);
  left2servo.attach(SERVO_LEFT_2);
  right1servo.attach(SERVO_RIGHT_1);
  right2servo.attach(SERVO_RIGHT_2);

}

/**
 * @brief do the moon walk function
 * 
 */
void moon_walk(){
 
  left2servo.write(flatPos + left2Offset - footTilt);
  right2servo.write(flatPos + right2Offset + footTilt);
  left1servo.write(flatPos + left1Offset);
  right1servo.write(flatPos + right1Offset);
  for (pos = startPos; pos <= endPos; pos += 1) { // goes from 0 degrees to 180 degrees
    // in steps of 1 degree
    //left1servo.write(pos);
    left2servo.write(pos);
    //right1servo.write(pos);
    right2servo.write(endPos + (startPos - pos));
    delay(15);                       // waits 15ms for the servo to reach the position
  }
  left1servo.write(flatPos + left1Offset + footTilt);
  right1servo.write(flatPos + right1Offset - footTilt);
  
  for (pos = endPos; pos >= startPos; pos -= 1) { // goes from 180 degrees to 0 degrees
    //left1servo.write(pos);
    left2servo.write(pos);
     //right1servo.write(pos);
    right2servo.write(endPos + (startPos - pos));            // tell servo to go to position in variable 'pos'
    delay(15);                       // waits 15ms for the servo to reach the position
  }


}

File: Testing/dance_test/moon_walk.py
"""
This file contains the defintion for the first dance we created called moon_walk()
"""
import time
import board
import pwmio
from adafruit_motor import servo

SERVO_LEFT_1 = pwmio.PWMOut(board.D12, duty_cycle=2 ** 15, frequency=50)
SERVO_LEFT_2 = pwmio.PWMOut(board.D11, duty_cycle=2 ** 15, frequency=50)
SERVO_RIGHT_1 = pwmio.PWMOut(board.D13, duty_cycle=2 ** 15, frequency=50)
SERVO_RIGHT_2 =pwmio.PWMOut(board.A1, duty_cycle=2 ** 15, frequency=50)

startPos = 40
pos = startPos    
endPos = 120
flatPos = 90
left1Offset = 10
left2Offset = 0
right1Offset = 5
right2Offset = 0
footTilt = 8

left1servo = servo.Servo(SERVO_LEFT_1)
left2servo = servo.Servo(SERVO_LEFT_2)
right1servo = servo.Servo(SERVO_RIGHT_1)
right2servo = servo.Servo(SERVO_RIGHT_2)

def moon_walk(walking:bool):
  while walking:
    left2servo.angle = (flatPos + left2Offset - footTilt)
    right2servo.angle = (flatPos + right2Offset + footTilt)
    left1servo.angle = (flatPos + left1Offset)
    right1servo.angle = (flatPos + right1Offset)
    for pos in range(startPos, endPos):
      left2servo.angle = (pos)
      right2servo.angle = (endPos + (startPos - pos))
      time.sleep(15/1000)
    
    left1servo.angle = (flatPos + left1Offset + footTilt)
    right1servo.angle = (flatPos + right1Offset - footTilt)
    
    for pos in range(endPos, startPos):
      left2servo.angle = (pos)
      right2servo.angle = (endPos + (startPos - pos));            
      time.sleep(15/1000);                       
  
moon_walk(True)
    

